app/cdap/components/WranglerGrid/SelectColumnPanel/DataTable/index.tsx

```diff
@@ -0,0 +1,193 @@
+  &.MuiTableCell-head {
+    ${cellFontCSS}
+    padding: 0;
+    color: ${grey[900]};
+    border-bottom: none !important;
+  }
+`;
+
+export default function({
+  columns,
+  transformationDataType,
+  handleSingleSelection,
+  selectedColumns,
+  dataQualityValue,
+  isSingleSelection,
+  handleDisableCheckbox,
+  handleMultipleSelection,
+  setSelectedColumns,
+  transformationName,
+}: IDataTableProps) {
+  const indexOfMultiSelectOption = MULTI_SELECTION_COLUMN.findIndex(
+    (option) => option.value === transformationName && option.isMoreThanTwo === false
+  );
+
+  // This function is used to either select all checkbox or uncheck the selected once
+  const handleChange = () => {
+    if (indexOfMultiSelectOption > -1) {
+      // If only two column selection is possible
+      if (selectedColumns.length) {
+        setSelectedColumns([]); // If columns are already selected then those all will be unchecked
+      } else {
+        if (columns.length > 2) {
+          setSelectedColumns(columns.slice(0, 2)); // When clicked on a checkbox then first two column will be selected in case if transformation present is join/swap
+        } else {
+          setSelectedColumns(columns); // When clicked on a checkbox then all columns will be selected
+        }
+      }
+    } else {
+      if (selectedColumns.length) {
+        setSelectedColumns([]); // If columns are already selected then those all will be unchecked
+      } else {
+        setSelectedColumns(columns); // When clicked on a checkbox then all columns will be selected
+      }
+    }
+  };
+
+  // This function is used to filter column based on their datatype which matches with the supported types of transformation to be applied
+  const getColumnsToDisplay = () => {
+    if (transformationDataType.includes('all')) {
+      return columns;
+    }
+    return columns.filter((column) =>
+      transformationDataType.includes(column.type[0].toLowerCase())
+    );
+  };
+
+  const columnsToDisplay: IHeaderNamesList[] = getColumnsToDisplay();
+
+  return (
+    <StyledTableContainer data-testid="column-table-parent">
+      {columnsToDisplay.length === 0 && (
+        <FlexWrapper>
+          <CenterAlignWrapper>
+            {NoDataSVG}
+            <SubHeadBoldFont component="p" data-testid="no-column-title">
+              {T.translate(`${SELECT_COLUMN_LIST_PREFIX}.noColumns`)}
+            </SubHeadBoldFont>
+            <NormalFont component="p" data-testid="no-column-subTitle">
+              {T.translate(`${SELECT_COLUMN_LIST_PREFIX}.noMatchColumnDatatype`)}
+            </NormalFont>
+          </CenterAlignWrapper>
+        </FlexWrapper>
+      )}
+      {columnsToDisplay.length > 0 && (
+        <StyledTable aria-label="recipe steps table">
+          <Divider />
+          <StyledTableHead>
+            <StyledTableRow>
+              <StyledTableHeadCell>
+                {MULTI_SELECTION_COLUMN.findIndex((option) => option.value === transformationName) >
```

Comment:
For testing if an element is in an array, you can use `array.some()` or even `!!array.find()`. Then you don't need an extra comparison with `-1`.

Reply:
There are still a number of `findIndex` calls in the PR

Reply:
@njbriggs I have updated the code
removed the findIndex wherever unnecessary
replaced it with .some and renamed a couple of variable for readability